vmm/src/api/http/mod.rs

```diff
@@ -41,39 +40,31 @@ pub mod http_endpoint;
 pub type HttpApiHandle = (thread::JoinHandle<Result<()>>, EventFd);
 
 /// Errors associated with VMM management
-#[derive(Debug)]
+#[derive(Error, Debug)]
 pub enum HttpError {
     /// API request receive error
-    SerdeJsonDeserialize(SerdeError),
+    #[error("API request receive error: {0}")]
+    SerdeJsonDeserialize(#[source] SerdeError),
 
     /// Attempt to access unsupported HTTP method
+    #[error("bad request: perhaps attempted to access an unsupported HTTP method?")]
     BadRequest,
 
     /// Undefined endpoints
+    #[error("endpoint not found")]
     NotFound,
 
     /// Too many requests
+    #[error("too many requests")]
     TooManyRequests,
 
     /// Internal Server Error
+    #[error("internal server error")]
     InternalServerError,
 
     /// Error from internal API
-    ApiError(ApiError),
-}
-
-impl Display for HttpError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        use self::HttpError::*;
-        match self {
-            BadRequest => write!(f, "Bad Request"),
-            NotFound => write!(f, "Not Found"),
-            TooManyRequests => write!(f, "Too Many Requests"),
-            InternalServerError => write!(f, "Internal Server Error"),
-            SerdeJsonDeserialize(serde_error) => write!(f, "{serde_error}"),
-            ApiError(api_error) => write!(f, "{api_error}"),
-        }
-    }
+    #[error("Error from internal API: {0:?}")]
```

Comment:
Uhm, I'm not entirely sure what is going on, but it looks like the test succeeds if I use the Display impl instead of Debug here:

```
    #[error("Error from internal API: {0}")]
```

Reply:
I cannot beliebe that the CI is actually green 😁 I'd feel uncomfortable if we merge this without
- understanding the root cause
- improve the debug experience for the next time

@rbradford

Reply:
I didn't have a chance to look at the code and try and reproduce manually - but I was wondering if there may have been a stray new line in the output and thus breaking the HTTP response?

Reply:
The other thing is that it could have been trying to do a stack unwind for the embedded debug error message which will then fail as the seccomp rules won't allow it.

Reply:
Hahahaha you were right, it is a seccomp violation. I just verified this by passing `--seccomp false` to the CHV command line. So the debugging guide here is: Run the test without seccomp if it suspicially fails.